### DISCUSSION TITLE
Funkverbindung zu HomematicIP Komponenten rein über HMIP-AP ohne installiertes Funkmodul

### DISCUSSION THREAD

**Discussion by MattN82** [OpenCCU/OpenCCU#1781] [🇺🇸 Q&A] [answered]
### Describe the issue you are experiencing

(EDIT: Weiß der Geier, warum ich das auf englisch geschrieben habe....die englische Menüführung wahrscheinlich :-) )

Hi,

my raspberrymatic runs on a VM inside of Virtual Machine Manager on a Synology NAS.
A HB-RF-USB-TK with an RPI-RF-MOD handles communication with the HMIP components in my System.

Recently I've added a HMIP-AP as a LAN gateway to increase the range of my HMIP "network".

As th HB-RF-USB-TK with RPI-RF-MOD is a kind of...diy solution (and a possible single point of failure)..I tried to run my raspberrymatic without it as an experiment....thinking it wouldn't be necessary as the HMIP-AP should just act as a kind of replacement. (so I thought). But it doesn't.

As soon as I boot up my rpm without the above mentioned USB-Module, it doesn't work.

Why? Is there any technical reason behind that? HMIP components can and do communicate with an AP exclusively (my garage is living proof of that :-D ). So why does RPM just....stop working as soon as there is no module installed instead of just "rerouting" the "traffic" over the HMIP-AP?

Thanks in advance

### Describe the behavior you expected

In my humble opinion, RPM should take advantage of the given RF-Hardware no matter what it is. The need for a centralized solution creates the risk of the whole installation failing if the module attached to the hardware (NAS in my case) stops working.

Whereas a system with multiple HMIP-APs could handle a single failure...even of the main module...as they would just "take over" the traffic.

### Steps to reproduce the issue

1. Take the Hardware I described (USB-module + HMIP-AP) -> Everything works fine
2. remove the main module but leave the HMIP-AP attached -> System won't even show duty cycle, as there is no RF going on
4.
...

### What is the version this bug report is based on?

3.63.8.20220330

### Which base platform are you running?

ova (Open Virtual Infrastructure)

### Which HomeMatic/homematicIP radio module are you using?

RPI-RF-MOD

### Anything in the logs that might be useful for us?

```txt
I feel like this is probably me misunderstanding some technical restraints...so I'm pretty much sure I will get the answer "this is normal":-)...so at this point I would hesitate to dive into logs...but I will surely provide them if this is NOT normal behavior
```

**Comment by jens-maus** [comment, marked as answer]
You are right, the answer is: "this is normal". In fact, a `RPI-RF-MOD` or `HmIP-RFUSB` RF module is required for using/adding a `HmIP-HAP`  as a range extender. The reason is, that these rf modules are not only doing rf communication, they are actually (and named so) co-processors for providing the cryptographic functionality for homematicIP communication in general. So a valid `RPI-RF-MOD` or `HmIP-RFUSB` is required for communication with a `HmIP-HAP` or even to/from a `HmIPW-DRAP` gateway for homematicIP-Wired communication.

So this is not a limitation, but a feature actually and the reason why in the documentation (see https://github.com/jens-maus/RaspberryMatic/wiki/Einleitung#voraussetzungen) they are listed as required.

**Reply by janosch74** [reply]
>...a `RPI-RF-MOD` or `HmIP-RFUSB` RF module is required for using/adding a `HmIP-HAP` as a range extender. 

Eine Rückfrage dazu: Mit der aktuellen Firmware sollte es doch möglich sein, des HmIP-HAP als LAN-Gateway zu konfigurieren. Ist die Aussage mit den zusätzlichen USB Funk-Modulen weiterhin gültig? Hintergrund: Meine Synology steht Funktechnisch sehr ungünstig und ich würde gerne auch aus anderen Gründen gerne auf die USB-Module verzichten und stattdessen ausschließlich HmIP-HAP als LAN-Gateway nutzen.
Geht das mit der neuen Firmware nun "standalone"?

**Reply by jens-maus** [reply]
> Hintergrund: Meine Synology steht Funktechnisch sehr ungünstig und ich würde gerne auch aus anderen Gründen gerne auf die USB-Module verzichten und stattdessen ausschließlich HmIP-HAP als LAN-Gateway nutzen. Geht das mit der neuen Firmware nun "standalone"?

Nein, das geht mit der aktuellen Firmware des HmIP-HAP nicht und wir auch vmtl. nie gehen. Ein HmIP-HAP braucht *immer* im Betrieb mit einer CCU/RaspberryMatic *zusätzlich* ein Funk/Kryptomodul wie ein RPI-RF-MOD oder HmIP-RFUSB.

**Comment by MattN82** [comment]
First of all: Thanks for the ultra-fast response!

I get that....sad for my showerthought-use-case....but I get it.

But that leads me to an additional question:

During my experiments, (which I didn't mention in my first post) I also tried replacing the existing "HB-RF-USB-TK with an RPI-RF-MOD"-Solution with a HMIP-RFUSB Stick.

Which caused the exact same behavior. (No connection to anything including the HMIP-AP)

Why is that?

**Reply by jens-maus** [reply]
A different problem of course!

**Comment by MattN82** [comment]
I See....so a primary module can neither be removed nor replaced
